io_ogre/ogre/skeleton.py

```diff
@@ -374,12 +376,32 @@ def write_animation( self, arm, actionName, frameBegin, frameEnd, doc, parentEle
             if bone.shouldOutput:
                 bone_tracks.append( Bone_Track(bone) )
             bone.clear_pose_transform()  # clear out any leftover pose transforms in case this bone isn't keyframed
-        for frame in range( int(frameBegin), int(frameEnd)+1, bpy.context.scene.frame_step):#thanks to Vesa
+        
+        # Only export keyframes (Exported animation won't be affected by Inverse Kinematics, Drivers and modified F-Curves)
+        key_frame_numbers = [] # Holds a list of keyframes
+        if config.get('ONLY_KEYFRAMES'):
+            action = bpy.data.actions[actionName] # actionName is the animation name (NLAtrack child)
+            # loops through all channels on the f-curve --> Code taken from: https://blender.stackexchange.com/questions/8387/how-to-get-keyframe-data-from-python
+            for fcu in action.fcurves:
+                for keyframe in fcu.keyframe_points: # Loops through all the keyframes in the channel
+                    kf = keyframe.co[0] # key frame number
+                    if kf not in key_frame_numbers: # only add the key frames in once
+                        key_frame_numbers.append(kf)
+            
```

Comment:
```suggestion
            else:
```

Reply:
What is the purpose of the else:?

Reply:
I think the code would be easier to read if structured as:
```py
if config.get('ONLY_KEYFRAMES'):
    frame_range = ...
else:
    frame_range = ...
```

Reply:
so the code looks like this:
```
# Only export keyframes (Exported animation won't be affected by Inverse Kinematics, Drivers and modified F-Curves)
key_frame_numbers = [] # Holds a list of keyframes
if config.get('ONLY_KEYFRAMES'):
    action = bpy.data.actions[actionName] # actionName is the animation name (NLAtrack child)
    # loops through all channels on the f-curve --> Code taken from: https://blender.stackexchange.com/questions/8387/how-to-get-keyframe-data-from-python
    for fcu in action.fcurves:
        for keyframe in fcu.keyframe_points: # Loops through all the keyframes in the channel
            kf = keyframe.co[0] # key frame number
            if kf not in key_frame_numbers: # only add the key frames in once
                key_frame_numbers.append(kf)
    
# Set the frame_range collection
if config.get('ONLY_KEYFRAMES'): # Modify frame range so only key frames are exported
    frame_range = key_frame_numbers 
else
    # Export keyframes over the begin-end range so that the exported animation will be affected by Inverse Kinematics, Drivers and modified F-Curves
    # frame_step is [( # of frames / FPS ) / # of frames ] -- I think
    frame_range = range( int(frameBegin), int(frameEnd)+1, bpy.context.scene.frame_step) #thanks to Vesa

# Add keyframes to export
for frame in frame_range:
    bpy.context.scene.frame_set(frame)
    for bone in self.roots:
        bone.update()
    for track in bone_tracks:
        track.add_keyframe((frame - frameBegin) / _fps)
```

Reply:
Also I just noticed issue #87. I haven't looked yet, but if this code is easily portable, do you want an option for node animation to export only key frames or be the ONLY option?

Reply:
```py
# Only export keyframes (Exported animation won't be affected by Inverse Kinematics, Drivers and modified F-Curves)
if config.get('ONLY_KEYFRAMES'):
    frame_range = []
    action = bpy.data.actions[actionName] # actionName is the animation name (NLAtrack child)
    # loops through all channels on the f-curve --> Code taken from: https://blender.stackexchange.com/questions/8387/how-to-get-keyframe-data-from-python
    for fcu in action.fcurves:
        for keyframe in fcu.keyframe_points: # Loops through all the keyframes in the channel
            kf = keyframe.co[0] # key frame number
            if kf not in key_frame_numbers: # only add the key frames in once
                frame_range.append(kf)
else
    # Export keyframes over the begin-end range so that the exported animation will be affected by Inverse Kinematics, Drivers and modified F-Curves
    # frame_step is [( # of frames / FPS ) / # of frames ] -- I think
    frame_range = range( int(frameBegin), int(frameEnd)+1, bpy.context.scene.frame_step) #thanks to Vesa

# Add keyframes to export
for frame in frame_range:
    bpy.context.scene.frame_set(frame)
    for bone in self.roots:
        bone.update()
    for track in bone_tracks:
        track.add_keyframe((frame - frameBegin) / _fps)
```